### ISSUE TITLE
Kernel crashes when executing code using tensorflow 

### ISSUE THREAD

**Issue by cosminborsa** [microsoft/vscode-jupyter#14902]
I'm trying to build a simple feed-forward neural network using TensorFlow (version 2.9.0). I load some arrays with input and output data, and when I try to train the network I get this error:

The Kernel crashed while executing code in the the current cell or a previous cell. Please review the code in the cell(s) to identify a possible cause of the failure. Click [here](https://aka.ms/vscodeJupyterKernelCrash) for more info. View Jupyter [log](command:jupyter.viewOutput) for further details.

When you look at the logs you see this:
17:11:03.414 [error] Disposing session as kernel process died ExitCode: 3221225725, Reason: 2023-12-11 17:07:20.606356: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX AVX2

I read on blogs that this shouldn't be an error, only a warning, but it appears that it causes the Jupyter kernel to crash. If I use the Spyder IDE without Jupyter, the code runs as usual.

## Environment data

-   VS Code version: 1.85.0 (user setup)
-   Jupyter Extension version (available under the Extensions sidebar): v2023.11.1003402403
-   Python Extension version (available under the Extensions sidebar): v2023.22.0
-   OS (Windows | Mac | Linux distro) and version: Windows 11
-   Python and/or Anaconda version: 3.10.13
-   Type of virtual environment used (N/A | venv | virtualenv | conda | ...): conda
-   Jupyter server running: Local 

## Expected behaviour

No errors

## Actual behaviour

Crash due to warning

## Steps to reproduce:

[**NOTE**: Self-contained, minimal reproducing code samples are **extremely** helpful and will expedite addressing your issue]

## Initializing the Neural Network
epnum = 5
opt = tf.keras.optimizers.Adam(learning_rate=0.001, beta_1=0.95, beta_2=0.999)
NN_agg_m2 = construction(mcall2, scaler_m2, x = x_m2, y = y_m2, node_num = 30, batch_num = 64, epoch_num = epnum, 
                         optim = opt, display_time=False)

import utils.utils as ut --- this is a python script that I wrote

## Libraries
import time
import numpy as np
import tensorflow as tf
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense
from tensorflow.keras.initializers import TruncatedNormal
       
def construction(model, scalar, x = None, y = None, node_num = 30, epoch_num = 5, batch_num = 64, actfct = 'relu', 
                    initializer = TruncatedNormal(mean = 0.0, stddev = 0.05, seed = 1), optim = 'adam', 
                    lossfct = 'mean_squared_error', display_time = False):
    '''
    Builds a single aggreagte neural network from input and output training pairs

    Parameters
    ----------
    model : Dictionary containing all the parameters of the stock and contract
    scalar : tuple
        A tuple containing a list of input scaling objects (size model['dim']+1) and an output scaling object.
    x : Array of input data used in training the aggregate neural network
    y : Array of output data used in training the aggregate neural network
    node_num : Integer with the number of nodes. The default is 30.
    epoch_num : Integer with the number of epochs. The default is 5.
    batch_num : Integer with the number of batches. The default is 64.
    actfct : String with the activation function. The default is 'relu'.
    initializer : Keras initializer. The default is set to 
                  TruncatedNormal(mean = 0.0, stddev = 0.05).
    optim : Keras optimizer. The default is 'adam'.
    lossfct : String stating the type of the loss function. The default is 
              'mean_squared_error'.
    display_time : Boolean asserting whether to display the time spent per step. 
                   The default is False.

    Returns
    -------
    NNagg : Aggregate Neural Network object
    '''
    # Set seed
    tf.random.set_seed(15)

    (convert_in, convert_out) = scalar
    if display_time:
        time_all = time.time()
    nn_dim_agg = model['dim'] + 1    # Dimension of the Aggregate Neural Network
    
    # Random suhffling of data points
    merge = np.append(x.transpose(), y.transpose()).reshape(model['dim']+2, len(y)).transpose()
    np.random.shuffle(merge)
    x = merge.transpose()[:-1].transpose()
    y = merge.transpose()[-1].reshape(-1, 1)
    
    # Scaling neural network inputs and outputs    
    input_scaled = []
    for j in range(nn_dim_agg):
            input_train = x.transpose()[j].reshape((-1, 1))
            input_scaled.append(convert_in[j].transform(input_train))
    nn_input = np.stack(input_scaled).transpose()[0]
    nn_output = convert_out.transform(y.reshape(-1,1))
    
    # Define and train the aggeegate neural network
    NNagg = Sequential()    
    NNagg.add(Dense(node_num, input_shape = (nn_dim_agg,), activation = actfct, \
                       kernel_initializer = initializer, bias_initializer = initializer))            
    NNagg.add(Dense(node_num, activation = actfct, kernel_initializer = initializer, \
                       bias_initializer = initializer))
    NNagg.add(Dense(node_num, activation = actfct, kernel_initializer = initializer, \
                       bias_initializer = initializer))
    NNagg.add(Dense(1, activation = None, kernel_initializer = initializer, \
                       bias_initializer = initializer)) 
    NNagg.compile(optimizer = optim, loss = lossfct, metrics=['accuracy'])
    NNagg.fit(nn_input, nn_output, epochs = epoch_num, batch_size = batch_num, verbose = 0)   

    if display_time:
        print('Training time:', np.round(time.time()-time_all, 2), 'sec')
        
    return NNagg

## Logs

17:01:21.428 [info] Restarted 60e325d7-1ff3-448f-a5e4-c194ed40fcd7
17:02:54.663 [info] Handle Execution of Cells 1 for ~\Git Repos\Deep-Learning-for-OSP\Diagnostics.ipynb
17:02:54.672 [info] Kernel acknowledged execution of cell 1 @ 1702342974671
17:02:54.852 [info] End cell 1 execution after 0.18s, completed @ 1702342974851, started @ 1702342974671
17:02:57.184 [info] Handle Execution of Cells 2 for ~\Git Repos\Deep-Learning-for-OSP\Diagnostics.ipynb
17:02:57.192 [info] Kernel acknowledged execution of cell 2 @ 1702342977192
17:03:11.889 [info] End cell 2 execution after 14.697s, completed @ 1702342991889, started @ 1702342977192
17:03:20.880 [info] Handle Execution of Cells 4 for ~\Git Repos\Deep-Learning-for-OSP\Diagnostics.ipynb
17:03:20.891 [info] Kernel acknowledged execution of cell 4 @ 1702343000890
17:03:21.517 [info] End cell 4 execution after 0.627s, completed @ 1702343001517, started @ 1702343000890
17:03:22.690 [info] Handle Execution of Cells 5 for ~\Git Repos\Deep-Learning-for-OSP\Diagnostics.ipynb
17:03:22.701 [info] Kernel acknowledged execution of cell 5 @ 1702343002700
17:03:31.699 [info] End cell 5 execution after 8.999s, completed @ 1702343011699, started @ 1702343002700
17:05:37.920 [info] Handle Execution of Cells 7 for ~\Git Repos\Deep-Learning-for-OSP\Diagnostics.ipynb
17:05:37.989 [info] Kernel acknowledged execution of cell 7 @ 1702343137989
17:05:42.130 [info] End cell 7 execution after 4.14s, completed @ 1702343142129, started @ 1702343137989
17:06:10.947 [info] Handle Execution of Cells 7 for ~\Git Repos\Deep-Learning-for-OSP\Diagnostics.ipynb
17:06:10.957 [info] Kernel acknowledged execution of cell 7 @ 1702343170957
17:06:11.427 [info] End cell 7 execution after 0.47s, completed @ 1702343171427, started @ 1702343170957
17:06:17.454 [info] Handle Execution of Cells 8 for ~\Git Repos\Deep-Learning-for-OSP\Diagnostics.ipynb
17:06:17.478 [info] Kernel acknowledged execution of cell 8 @ 1702343177477
17:06:18.019 [info] End cell 8 execution after 0.542s, completed @ 1702343178019, started @ 1702343177477
17:06:24.604 [info] Handle Execution of Cells 8 for ~\Git Repos\Deep-Learning-for-OSP\Diagnostics.ipynb
17:06:24.616 [info] Kernel acknowledged execution of cell 8 @ 1702343184616
17:06:25.031 [info] End cell 8 execution after 0.415s, completed @ 1702343185031, started @ 1702343184616
17:06:31.699 [info] Handle Execution of Cells 8 for ~\Git Repos\Deep-Learning-for-OSP\Diagnostics.ipynb
17:06:31.708 [info] Kernel acknowledged execution of cell 8 @ 1702343191707
17:06:32.298 [info] End cell 8 execution after 0.591s, completed @ 1702343192298, started @ 1702343191707
17:06:35.048 [info] Handle Execution of Cells 9 for ~\Git Repos\Deep-Learning-for-OSP\Diagnostics.ipynb
17:06:35.081 [info] Kernel acknowledged execution of cell 9 @ 1702343195081
17:06:35.786 [info] End cell 9 execution after 0.705s, completed @ 1702343195786, started @ 1702343195081
17:06:42.508 [warn] Timeout (after 20000ms) waiting to inspect code 'x_m2.T'
17:06:42.635 [warn] Timeout (after 20000ms) waiting to inspect code 'x_m2.searchsorted'
17:06:42.748 [warn] Timeout (after 20000ms) waiting to inspect code 'x_m2.shape'
17:07:05.493 [info] Handle Execution of Cells 7 for ~\Git Repos\Deep-Learning-for-OSP\Diagnostics.ipynb
17:07:05.501 [info] Kernel acknowledged execution of cell 7 @ 1702343225501
17:07:06.325 [info] End cell 7 execution after 0.824s, completed @ 1702343226325, started @ 1702343225501
17:07:08.616 [info] Handle Execution of Cells 9 for ~\Git Repos\Deep-Learning-for-OSP\Diagnostics.ipynb
17:07:08.623 [info] Kernel acknowledged execution of cell 9 @ 1702343228623
17:07:09.100 [info] End cell 9 execution after 0.476s, completed @ 1702343229099, started @ 1702343228623
17:07:12.592 [info] Handle Execution of Cells 8 for ~\Git Repos\Deep-Learning-for-OSP\Diagnostics.ipynb
17:07:12.596 [info] Kernel acknowledged execution of cell 8 @ 1702343232596
17:07:12.995 [info] End cell 8 execution after 0.399s, completed @ 1702343232995, started @ 1702343232596
17:07:15.363 [info] Handle Execution of Cells 8 for ~\Git Repos\Deep-Learning-for-OSP\Diagnostics.ipynb
17:07:15.367 [info] Kernel acknowledged execution of cell 8 @ 1702343235367
17:07:15.792 [info] End cell 8 execution after 0.425s, completed @ 1702343235792, started @ 1702343235367
17:07:18.116 [info] Handle Execution of Cells 10 for ~\Git Repos\Deep-Learning-for-OSP\Diagnostics.ipynb
17:07:18.123 [info] Kernel acknowledged execution of cell 10 @ 1702343238123
17:07:20.606 [warn] StdErr from Kernel Process 2023-12-11 17:07:20.606356: I te
17:07:20.607 [warn] StdErr from Kernel Process nsorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimize
17:07:20.608 [warn] StdErr from Kernel Process d with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance
17:07:20.608 [warn] StdErr from Kernel Process -critical operations:  AVX AVX2
To enable them in other operations, rebuild TensorFlow with the appropriate compile
17:07:20.609 [warn] StdErr from Kernel Process r flags.
17:07:22.797 [warn] StdErr from Kernel Process 2023-12-11 17:07:22.797055: I tensorflow/core/common_runtime/gpu/gpu_device.cc:15
17:07:22.798 [warn] StdErr from Kernel Process 32] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 2782 MB memory:  -> device: 0, name: NVIDIA GeForce GTX 1050 Ti, pci bus id: 0000:01:00.0, compute capability: 6.1
17:11:03.414 [error] Disposing session as kernel process died ExitCode: 3221225725, Reason: 2023-12-11 17:07:20.606356: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX AVX2

Packages in conda environment:

absl-py                   2.0.0                    pypi_0    pypi
asttokens                 2.4.1                    pypi_0    pypi
astunparse                1.6.3                    pypi_0    pypi
bzip2                     1.0.8                he774522_0
ca-certificates           2023.08.22           haa95532_0
cachetools                5.3.2                    pypi_0    pypi
certifi                   2023.11.17               pypi_0    pypi
charset-normalizer        3.3.2                    pypi_0    pypi
cloudpickle               3.0.0                    pypi_0    pypi
colorama                  0.4.6                    pypi_0    pypi
comm                      0.2.0                    pypi_0    pypi
contourpy                 1.2.0                    pypi_0    pypi
cycler                    0.12.1                   pypi_0    pypi
debugpy                   1.8.0                    pypi_0    pypi
decorator                 5.1.1                    pypi_0    pypi
exceptiongroup            1.2.0                    pypi_0    pypi
executing                 2.0.1                    pypi_0    pypi
flatbuffers               1.12                     pypi_0    pypi
fonttools                 4.46.0                   pypi_0    pypi
gast                      0.4.0                    pypi_0    pypi
google-auth               2.25.2                   pypi_0    pypi
google-auth-oauthlib      0.4.6                    pypi_0    pypi
google-pasta              0.2.0                    pypi_0    pypi
grpcio                    1.60.0                   pypi_0    pypi
h5py                      3.10.0                   pypi_0    pypi
idna                      3.6                      pypi_0    pypi
ipykernel                 6.27.1                   pypi_0    pypi
ipython                   8.18.1                   pypi_0    pypi
jedi                      0.19.1                   pypi_0    pypi
joblib                    1.3.2                    pypi_0    pypi
jupyter-client            8.6.0                    pypi_0    pypi
jupyter-core              5.5.0                    pypi_0    pypi
keras                     2.9.0                    pypi_0    pypi
keras-preprocessing       1.1.2                    pypi_0    pypi
kiwisolver                1.4.5                    pypi_0    pypi
libclang                  16.0.6                   pypi_0    pypi
libffi                    3.4.4                hd77b12b_0
markdown                  3.5.1                    pypi_0    pypi
markupsafe                2.1.3                    pypi_0    pypi
matplotlib                3.8.2                    pypi_0    pypi
matplotlib-inline         0.1.6                    pypi_0    pypi
nest-asyncio              1.5.8                    pypi_0    pypi
numpy                     1.26.2                   pypi_0    pypi
oauthlib                  3.2.2                    pypi_0    pypi
openssl                   3.0.12               h2bbff1b_0
opt-einsum                3.3.0                    pypi_0    pypi
packaging                 23.2                     pypi_0    pypi
parso                     0.8.3                    pypi_0    pypi
pillow                    10.1.0                   pypi_0    pypi
pip                       23.3.1          py310haa95532_0
platformdirs              4.1.0                    pypi_0    pypi
prompt-toolkit            3.0.41                   pypi_0    pypi
protobuf                  3.19.6                   pypi_0    pypi
psutil                    5.9.6                    pypi_0    pypi
pure-eval                 0.2.2                    pypi_0    pypi
pyasn1                    0.5.1                    pypi_0    pypi
pyasn1-modules            0.3.0                    pypi_0    pypi
pygments                  2.17.2                   pypi_0    pypi
pyparsing                 3.1.1                    pypi_0    pypi
python                    3.10.13              he1021f5_0
python-dateutil           2.8.2                    pypi_0    pypi
pywin32                   306                      pypi_0    pypi
pyzmq                     25.1.2                   pypi_0    pypi
requests                  2.31.0                   pypi_0    pypi
requests-oauthlib         1.3.1                    pypi_0    pypi
rsa                       4.9                      pypi_0    pypi
scikit-learn              1.3.2                    pypi_0    pypi
scipy                     1.11.4                   pypi_0    pypi
setuptools                68.0.0          py310haa95532_0
six                       1.16.0                   pypi_0    pypi
spyder-kernels            2.5.0                    pypi_0    pypi
sqlite                    3.41.2               h2bbff1b_0
stack-data                0.6.3                    pypi_0    pypi
tensorboard               2.9.1                    pypi_0    pypi
tensorboard-data-server   0.6.1                    pypi_0    pypi
tensorboard-plugin-wit    1.8.1                    pypi_0    pypi
tensorflow                2.9.0                    pypi_0    pypi
tensorflow-estimator      2.9.0                    pypi_0    pypi
tensorflow-io-gcs-filesystem 0.31.0                   pypi_0    pypi
termcolor                 2.4.0                    pypi_0    pypi
threadpoolctl             3.2.0                    pypi_0    pypi
tk                        8.6.12               h2bbff1b_0
tornado                   6.4                      pypi_0    pypi
traitlets                 5.14.0                   pypi_0    pypi
typing-extensions         4.9.0                    pypi_0    pypi
tzdata                    2023c                h04d1e81_0
urllib3                   2.1.0                    pypi_0    pypi
vc                        14.2                 h21ff451_1
vs2015_runtime            14.27.29016          h5e58377_2
wcwidth                   0.2.12                   pypi_0    pypi
werkzeug                  3.0.1                    pypi_0    pypi
wheel                     0.41.2          py310haa95532_0
wrapt                     1.16.0                   pypi_0    pypi
xz                        5.4.5                h8cc25b3_0
zlib                      1.2.13               h8cc25b3_0

<summary>Output for <code>Jupyter</code> in the <code>Output</code> panel (<code>View</code>→<code>Output</code>, change the drop-down the upper-right of the <code>Output</code> panel to <code>Jupyter</code>)
</summary>

<p>

```
XXX
```

</p>
</details>

**Comment by DonJayamanne** [comment]
> 17:11:03.414 [error] Disposing session as kernel process died ExitCode: 3221225725, Reason: 2023-12-11 17:07:20.606356: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations: AVX AVX2

I read on blogs that this shouldn't be an error, only a warning, but it appears that it causes the Jupyter kernel to crash. If I use the Spyder IDE without Jupyter, the code runs as usual.

Please can you verify you are running against the exact same python environment.
I.e. run the followihng code in spyder and vscode and verify the output is the same
```
import sys
print(sys.executable)
```

We have found that the keras and similar python packages can cause kernels to crash when not installed correctly. I suspect thats whats happening with this environment as well. Please do try the above and share the info.
thanks

**Comment by cosminborsa** [comment]
print(sys.executable) returns the same paths since I'm using the same kernel. Also Hello World works. 

I think I fixed the issue: VS Code was masking the actual error... which was a stack overflow. When I ran the same code in Spyder I got the warning: TensorFlow binary is optimized with oneAPI Deep Neural Network... but then the kernel crashed because of stack overflow in the code I had next. 

After fixing the stack overflow issue, the kernel works as usual.
